core/src/main/java/com/yahoo/oak/EntryArray.java

```diff
@@ -77,10 +53,9 @@
      */
     EntryArray(OakSharedConfig<K, V> config, int additionalFieldCount, int entriesCapacity) {
         this.config = config;
-        this.fields = additionalFieldCount + 2; // +2 for key and value references that always exist
-        this.entries = new long[entriesCapacity * this.fields];
+        // +2 for key and value references that always exist
+        this.array = new EntryArrayHeap(entriesCapacity, additionalFieldCount + 2);
```

Comment:
May be to put `additionalFieldCount + 2` to a variable named `fieldsPerEntryCounter`, or to add some comment, to make it a bit clearer